lib/rules/require-default-parameters.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Requires use of default parameters instead of checking for undefined in the function body
+ *
+ * Type: `Boolean`
+ *
+ * Value: `true`
+ *
+ * Version: `ES6`
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireDefaultParameters": true
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * function multiply(a, b = 1) {}
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * function multiply(a, b) { b = b || 1; }
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+
+    configure: function(options) {
+        assert(
+            options === true,
+            this.getOptionName() + ' option requires a true value or should be removed'
+        );
+    },
+
+    getOptionName: function() {
+        return 'requireDefaultParameters';
+    },
+
+    check: function(file, errors) {
+        file.iterateNodesByType('AssignmentExpression', function(node) {
+            var left = node.left;
+            var right = node.right;
+            if (
+                // b = b || 1
+                right.type === 'LogicalExpression' &&
+                left.type === 'Identifier' && right.left.type === 'Identifier' &&
+                left.name === right.left.name ||
```

Comment:
We don't have a helper. We have node iteration by function types. However, if you need to have utility functions on a function node (like getting the list of params), my vote would be for creating `lib/Function` that is a class that takes in a node and computes the params list from it. It could then have utility methods on it for getting the params list.

It might feel like a yagni to create this abstraction, but `lib/utils` really shouldn't be a grab bag of random methods.